File: Artist/app/src/main/java/com/example/artist/ui/topalbumslisting/di/TopAlbumsComponent.java
package com.example.artist.ui.topalbumslisting.di;



import com.example.artist.ui.topalbumslisting.TopAlbumsFragment;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {TopAlbumsModule.class})
public interface TopAlbumsComponent {
    void inject(TopAlbumsFragment target);
}

File: Artist/app/src/main/java/com/example/artist/ui/topalbumslisting/di/TopAlbumsComponent.kt
package com.example.artist.ui.topalbumslisting.di

import com.example.artist.ui.topalbumslisting.TopAlbumsFragment
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [TopAlbumsModule::class])
interface TopAlbumsComponent {
    fun inject(target: TopAlbumsFragment)
}